go/consensus/tendermint/abci/mux.go

```diff
@@ -769,10 +771,18 @@ func (mux *abciMux) CheckTx(req types.RequestCheckTx) types.ResponseCheckTx {
 }
 
 func (mux *abciMux) DeliverTx(req types.RequestDeliverTx) types.ResponseDeliverTx {
-	ctx := mux.state.NewContext(ContextDeliverTx, mux.currentTime)
+	ctx := mux.state.NewContext(api.ContextDeliverTx, mux.currentTime)
 	defer ctx.Close()
 
 	if err := mux.executeTx(ctx, req.Tx); err != nil {
+		if api.IsUnavailableStateError(err) {
+			// Make sure to not commit any transactions which include results based on unavailable
+			// and/or corrupted state -- doing so can further corrupt state.
+			ctx.Logger().Error("unavailable and/or corrupted state detected during tx processing",
+				"err", err,
+			)
+			panic(err)
```

Comment:
Yeah, unfortunately ABCI does not provide a way to signal such failures -- the only way is to panic. This will cause a consensus failure and Commit will not be called so everything will be discarded.

Reply:
oh consensus commit. thanks